File: android/2048/src/main/java/com/dreampany/play2048/ui/activity/NavigationActivity.java
package com.dreampany.play2048.ui.activity;

import android.os.Bundle;

import com.dreampany.play2048.ui.fragment.HomeFragment;
import com.dreampany.framework.misc.SmartAd;
import com.dreampany.framework.ui.activity.BaseBottomNavigationActivity;
import com.dreampany.framework.ui.fragment.BaseFragment;
import com.dreampany.play2048.R;
import com.dreampany.play2048.ui.fragment.MoreFragment;

import javax.inject.Inject;

import dagger.Lazy;

/**
 * Created by Hawladar Roman on 5/24/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
public class NavigationActivity extends BaseBottomNavigationActivity {

    @Inject
    Lazy<HomeFragment> homeFragment;
    @Inject
    Lazy<MoreFragment> moreFragment;
    @Inject
    SmartAd ad;
//    ActivityNavigationBinding binding;

    @Override
    public int getLayoutId() {
        return R.layout.activity_navigation;
    }

    @Override
    public int getToolbarId() {
        return R.id.toolbar;
    }

    @Override
    public int getNavigationViewId() {
        return R.id.navigation_view;
    }

    @Override
    public int getDefaultSelectedNavigationItemId() {
        return R.id.item_home;
    }

    @Override
    public boolean isHomeUp() {
        return false;
    }

    @Override
    protected void onStartUi(Bundle state) {
       // binding = (ActivityNavigationBinding) super.binding;
        ad.loadBanner(findViewById(R.id.adview));
    }

    @Override
    public void onBackPressed() {
        BaseFragment fragment = getCurrentFragment();
        if (fragment != null && fragment.hasBackPressed()) {
            return;
        }
        finish();
    }

    @Override
    protected void onNavigationItem(int navigationItemId) {
        switch (navigationItemId) {
            case R.id.item_home:
                commitFragment(HomeFragment.class, homeFragment, R.id.layout);
                break;
            case R.id.item_more:
                commitFragment(MoreFragment.class, moreFragment, R.id.layout);
                break;
        }
    }

    @Override
    protected void onStopUi() {

    }
}

File: android/history/src/main/kotlin/com/dreampany/history/ui/activity/NavigationActivity.kt
package com.dreampany.history.ui.activity

import android.os.Bundle
import com.dreampany.history.R
import com.dreampany.history.databinding.ActivityNavigationBinding
import com.dreampany.history.misc.Constants
import com.dreampany.history.ui.fragment.HomeFragment
import com.dreampany.history.ui.fragment.MoreFragment
import com.dreampany.history.ui.model.UiTask
import com.dreampany.frame.misc.SmartAd
import com.dreampany.frame.ui.activity.BaseBottomNavigationActivity
import com.dreampany.frame.ui.callback.SearchViewCallback
import com.dreampany.history.ui.fragment.FavoriteFragment
import com.miguelcatalan.materialsearchview.MaterialSearchView
import dagger.Lazy
import javax.inject.Inject

/**
 * Created by Roman-372 on 7/24/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class NavigationActivity : BaseBottomNavigationActivity(), SearchViewCallback {

    @Inject
    internal lateinit var homeFragment: Lazy<HomeFragment>
    @Inject
    internal lateinit var favoriteFragment: Lazy<FavoriteFragment>
    @Inject
    internal lateinit var moreFragment: Lazy<MoreFragment>
    @Inject
    internal lateinit var ad: SmartAd

    private lateinit var bind: ActivityNavigationBinding

    override fun getLayoutId(): Int {
        return R.layout.activity_navigation
    }

    override fun getToolbarId(): Int {
        return R.id.toolbar
    }

    override fun getNavigationViewId(): Int {
        return R.id.navigation_view
    }

    override fun getDefaultSelectedNavigationItemId(): Int {
        return R.id.item_home
    }

    override fun isHomeUp(): Boolean {
        return false
    }

    override fun hasDoubleBackPressed(): Boolean {
        return true
    }

    override fun getScreen(): String {
        return Constants.navigation(applicationContext)
    }

    override fun onStartUi(state: Bundle?) {
        initView()
        ad.loadBanner(getScreen())
    }

    override fun onStopUi() {
        ad.destroyBanner(getScreen())
    }

    override fun onResume() {
        super.onResume()
        ad.resumeBanner(getScreen())
    }

    override fun onPause() {
        ad.pauseBanner(getScreen())
        super.onPause()
    }

    override fun onNavigationItem(navigationItemId: Int) {
        when (navigationItemId) {
            R.id.item_home -> commitFragment(HomeFragment::class.java, homeFragment, R.id.layout)
            R.id.item_favorite -> commitFragment(FavoriteFragment::class.java, favoriteFragment, R.id.layout)
            R.id.item_more -> commitFragment(MoreFragment::class.java, moreFragment, R.id.layout)
        }
    }

    override fun getSearchView(): MaterialSearchView {
        return bind.searchView
    }

    private fun initView() {
        bind = super.binding as ActivityNavigationBinding
        val uiTask = getCurrentTask<UiTask<*>>(false)
        if (uiTask != null && uiTask!!.type != null && uiTask!!.subtype != null) {
            openActivity(ToolsActivity::class.java, uiTask!!)
            return
        }
        ad.initAd(
            this,
            getScreen(),
            findViewById(R.id.adview),
            R.string.interstitial_ad_unit_id,
            R.string.rewarded_ad_unit_id
        )
    }
}